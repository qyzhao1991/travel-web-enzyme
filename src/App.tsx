import React from 'react';
import './App.css';
import {Header} from "antd/lib/layout/layout";
import {Menu} from "antd";
import TravelFees from "./views/TravelFees";
import 'antd/dist/antd.css';

function App() {
  const [current, setCurrent] = React.useState<string>('home');
  
  return (
    <>
      <Header style={{fontSize: 36, color: 'white'}}>Enterprise Travel Management System</Header>
      <Menu onClick={(menu) => setCurrent(menu.key)} selectedKeys={[current]} mode="horizontal">
        <Menu.Item key="home">
          Home Page
        </Menu.Item>
        <Menu.Item key="travelFees">
          Travel Fees
        </Menu.Item>
        <Menu.Item key="accountManagement">
          Account Management
        </Menu.Item>
      </Menu>
      {current === 'travelFees' && <TravelFees/>}
    </>
  
  );
}

export default App;
