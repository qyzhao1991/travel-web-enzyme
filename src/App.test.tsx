import React from 'react';
import {shallow} from "enzyme";
import App from "./App";

describe("test enzyme", () => {
  it('should render success', () => {
     const wrapper = shallow(<App />);
     expect(wrapper).toMatchSnapshot();
  });
})
