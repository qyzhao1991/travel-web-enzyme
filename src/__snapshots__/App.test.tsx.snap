// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test enzyme should render success 1`] = `
<Fragment>
  <Header
    style={
      Object {
        "color": "white",
        "fontSize": 36,
      }
    }
  >
    Enterprise Travel Management System
  </Header>
  <Menu
    mode="horizontal"
    onClick={[Function]}
    selectedKeys={
      Array [
        "home",
      ]
    }
  >
    <MenuItem
      key="home"
    >
      Home Page
    </MenuItem>
    <MenuItem
      key="travelFees"
    >
      Travel Fees
    </MenuItem>
    <MenuItem
      key="accountManagement"
    >
      Account Management
    </MenuItem>
  </Menu>
</Fragment>
`;

exports[`test enzyme should render travel fees component when click menu 1`] = `
<Fragment>
  <Header
    style={
      Object {
        "color": "white",
        "fontSize": 36,
      }
    }
  >
    Enterprise Travel Management System
  </Header>
  <Menu
    mode="horizontal"
    onClick={[Function]}
    selectedKeys={
      Array [
        "home",
      ]
    }
  >
    <MenuItem
      key="home"
    >
      Home Page
    </MenuItem>
    <MenuItem
      key="travelFees"
    >
      Travel Fees
    </MenuItem>
    <MenuItem
      key="accountManagement"
    >
      Account Management
    </MenuItem>
  </Menu>
</Fragment>
`;
